stubs/pyxdg/xdg/DesktopEntry.pyi

```diff
@@ -0,0 +1,67 @@
+import re
+from os import PathLike
+from typing import AnyStr
+from typing_extensions import Literal
+
+from xdg.IniFile import IniFile
+
+class DesktopEntry(IniFile):
+    defaultGroup: str
+    content: dict[str, dict[str, str]]
+    def __init__(self, filename: PathLike[AnyStr] | None = None) -> None: ...
```

Comment:
AnyStr doesn't make sense here because the class is not generic. I think you just want a union:
```suggestion
    def __init__(self, filename: PathLike[str | bytes] | None = None) -> None: ...
```
Same on the next line.

Reply:
Also, since `.new()` only supports str paths and this function delegates to new sometimes, maybe it's safer to also only allow str paths here.